### PR TITLE
Move SQLDelight to a Dedicated Module and Update Dependencies

### LINKED ISSUE #214: [refactor] update dependencies
Update project dependencies

### PR DESCRIPTION
**Pull request by VahidGarousi**
This PR moves SQLDelight logic into a dedicated `ComposeNews.data.sqldelight` module, isolating `.sq` file to improve modularity and maintainability. 

### Key Changes:
- Created a new module for SQLDelight, making it the API for database interactions.
- Updated `build.gradle.kts` and `settings.gradle.kts` to reflect the new structure.
- Refactored `MarketDaoImpl.kt` to align with the new setup.

### CODE REVIEW

File: build-logic/convention/src/main/kotlin/HiltConventionPlugin.kt [hunk 1]
```diff
@@ -14,16 +14,16 @@ class HiltConventionPlugin : Plugin<Project> {
     private fun Project.applyPlugins() {
         pluginManager.apply {
             apply("dagger.hilt.android.plugin")
-            apply("org.jetbrains.kotlin.kapt")
-//            apply("com.google.devtools.ksp")
+//            apply("org.jetbrains.kotlin.kapt")
```
Comment (Kaaveh) [inline build-logic/convention/src/main/kotlin/HiltConventionPlugin.kt]: Can this comment be removed?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# ComposeNews
+# ComposeNews new
```
Comment (Kaaveh) [inline README.md]: Why did you add `new`?

File: data/sqldelight/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Kaaveh) [inline data/sqldelight/src/main/AndroidManifest.xml]: I think this file can be deleted.

**Review by Kaaveh — CHANGES_REQUESTED** [review]

File: data/sqldelight/src/main/AndroidManifest.xml (hunk 3 above)
Comment (VahidGarousi) [reply data/sqldelight/src/main/AndroidManifest.xml]: Removed

File: README.md (hunk 2 above)
Comment (VahidGarousi) [reply README.md]: Fixed

File: build-logic/convention/src/main/kotlin/HiltConventionPlugin.kt (hunk 1 above)
Comment (VahidGarousi) [reply build-logic/convention/src/main/kotlin/HiltConventionPlugin.kt]: Removed

**Review by Kaaveh — APPROVED** [review]

**Pull request merged**